src/code/fv-components/bottom-hud.js

```diff
@@ -2,31 +2,57 @@ import React, {PropTypes} from 'react';
 import AvatarButtonView from './avatar-button';
 import LevelIndicatorView from './level-indicator';
 import CountView from './counter';
+import GemSetView from './gem-set';
 import t from '../utilities/translate';
 
-const BottomHUDView = ({level, trial, trialCount, currScore, maxScore, currMoves, goalMoves}) => {
-  let scoreView = maxScore ? <CountView countTitleText={t('~COUNTER.SCORE_LABEL')} className={'score-count'} currCount={currScore} maxCount={maxScore} /> : null,
-      movesView = goalMoves > 0 ? <CountView countTitleText={t('~COUNTER.MOVES_LABEL')} className={'moves-count'} currCount={currMoves} maxCount={goalMoves} /> : null;
-
-  return (
-    <div id='fv-bottom-hud' className='fv-hud fv-bottom-hud' >
-      <AvatarButtonView />
-      <LevelIndicatorView level={level} />
-      <CountView countTitleText={t('~COUNTER.TRIAL_LABEL')} className={'trial-count'} currCount={trial} maxCount={trialCount} />
-      {scoreView}
-      {movesView}
-    </div>
-  );
-};
-
-BottomHUDView.propTypes = {
-  level: PropTypes.number,
-  trial: PropTypes.number,
-  trialCount: PropTypes.number,
-  currScore: PropTypes.number,
-  maxScore: PropTypes.number,
-  currMoves: PropTypes.number,
-  goalMoves: PropTypes.number
-};
+const BottomHUDView = React.createClass({
+
+  propTypes: {
+    routeSpec: PropTypes.object,
+    trial: PropTypes.number,
+    trialCount: PropTypes.number,
+    currScore: PropTypes.number,
+    maxScore: PropTypes.number,
+    currMoves: PropTypes.number,
+    goalMoves: PropTypes.number,
+    challengeProgress: PropTypes.object,
+    numChallenges: PropTypes.number,
+    showAward: PropTypes.bool
+  },
+
+  getInitialState() {
+    return {};
+  },
+
+  componentWillReceiveProps(nextProps) {
+    const { routeSpec: prevRouteSpec } = this.props,
+          { level: prevLevel, mission: prevMission, challenge: prevChallenge } = prevRouteSpec,
+          { routeSpec: nextRouteSpec } = nextProps,
+          { level: nextLevel, mission: nextMission, challenge: nextChallenge } = nextRouteSpec,
+          newChallenge = (prevLevel !== nextLevel) || (prevMission !== nextMission) || (prevChallenge !== nextChallenge);
+
+    // Only update the visible gems on a new level, or whenever a level finishes
+    if (newChallenge || this.currentProgress === {} || nextProps.showAward) {
+      this.setState({currentProgress: nextProps.challengeProgress});
+    }
```

Comment:
This is the fix for the bug that popped up during the demo, where gems would update mid-trial. Now, the gems in the HUD only change when a new level starts, or when a level finishes.